From ba414269013735b53a32efc1a97e015c9175c8ae Mon Sep 17 00:00:00 2001
From: Hemanth Kumar <hemanth@alifsemi.com>
Date: Mon, 1 Dec 2025 16:30:01 +0530
Subject: [PATCH 5/5] Skip vcvtaq_s32_f32 for AArch32

For some reason vcvtaq_s32_f32 is treated as undefined for AArch32 so rather
use alternative code.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hemanth Kumar <hemanth@alifsemi.com>
---
 tensorflow/lite/kernels/internal/optimized/neon_tensor_utils.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tensorflow/lite/kernels/internal/optimized/neon_tensor_utils.cc b/tensorflow/lite/kernels/internal/optimized/neon_tensor_utils.cc
index f8e7601f6b4..36504648721 100644
--- a/tensorflow/lite/kernels/internal/optimized/neon_tensor_utils.cc
+++ b/tensorflow/lite/kernels/internal/optimized/neon_tensor_utils.cc
@@ -2345,7 +2345,7 @@ void NeonVectorScalarMultiply(const int8_t* vector, const int v_size,
 // TODO(b/185850916): Consider changing the rounding stragey from "ties to away"
 // to "ties to even" since vcvtnq_s32_f32 is generally more available.
 inline int32x4_t RoundToNearest(const float32x4_t input) {
-#if __ARM_ARCH >= 8
+#if __ARM_ARCH >= 8 && !defined(__arm__)
   return vcvtaq_s32_f32(input);
 #else
   static const float32x4_t zero_val_dup = vdupq_n_f32(0.0f);
--
2.43.0

